src/gui/main.cpp

```diff
@@ -352,8 +321,59 @@ QSettings::setDefaultFormat(CkbSettings::Format);
         // If launched with --background, launch in background
         background = true;
         break;
+    case CommandLineDaemon:
+        // If launched with --daemon, launch as daemon
+        background = true;
+        daemon = true;
+        break;
+    }
     }
 
+    // Setup main application
+    std::unique_ptr<QCoreApplication> app
+      (daemon ? new QCoreApplication(argc, argv) : new QApplication(argc, argv));
+    QCoreApplication& a = *app;
+
+    // Setup translations
+    QTranslator translator;
+    if(translator.load(QLocale(), "", "", ":/translations"))
+        a.installTranslator(&translator);
+
+    const quint16 currentSettingsVersion = tmpSettings->value("Program/SettingsVersion", 0).toInt();
+    if(currentSettingsVersion && currentSettingsVersion > CKB_NEXT_SETTINGS_VER){
+        if(QMessageBox::warning(nullptr, QObject::tr("Downgrade Warning"),
```

Comment:
This should straight up print an error and quit if there's going to be a downgrade in daemon mode.